### DISCUSSION TITLE
m3u8 playlists import but have no tacks

### DISCUSSION THREAD

**Discussion by odjhl88** [navidrome/navidrome#2291] [Q&A] [answered]
Hello,
I am admittedly new to Navidrome, but everything works correctly, including an SSL through a Reverse Proxy, with no issues playing media. I have an NFS mount from a VM on my Prroxmox server connected to my NAS under `/mnt/cloud-music/<Nas Music Library Folders>`.
As I said, I have no issues playing anything in Navidrome from the web app or other applications.
I also set the playlists folder in my toml config with playlist auto import enabled.
The playlist folder exists within the music library folder as `_00Playlists`.
The playlists appear in the web app, but it doesn't place any of the songs within.
I initially suspected it was because the playlists were originally used in Foobar2000 and had a different file path associated with them, and the Windows file structure of using `\`s rather than the Debian Linux file structure of `/`s.
Understanding that the music is relative to the new music folder path, I changed all the file paths to the local path after the `cloud-music` mount folder and switched the sub-directory characters from `\` to `/`.
Please take a look at my sample m3u8 playlist structure below.
Can anyone explain why Navidrome can't associate the correct songs with the playlist?
Kind regards, odjhl88
```
#
Various Artists/70's Country [Madacy]/02 (Don't Take Her) She's All I Got.mp3
Various Artists/70's Country [Madacy]/08 9,999,999 Tears.mp3
Various Artists/70's Country [Madacy]/10 Country Bumpkin.mp3
Various Artists/70's Country [Madacy]/06 Lonely Women Make Good Lovers.mp3
Aaron Lewis/The Road/75.mp3
Aaron Lewis/Frayed At Both Ends/Again.mp3
Aaron Lewis/Frayed At Both Ends/Am I The Only One.mp3
Aaron Lewis/The Road/Anywhere but Here.mp3
```
![image](https://user-images.githubusercontent.com/86865288/229330236-29c9dab4-3fd5-4a31-bfb4-724e2f1c1b6f.png)

**Comment by deluan** [comment, marked as answer]
Paths inside the `m3u` files have to be relative to the `m3u` file itself. In your example above, it would only work if the `m3u` file was in the same folder as `Various Artists` and `Aaron Lewis` folders.

**Reply by odjhl88** [reply]
Perfect, I just switched the file path to go back a folder layer and then look for the artist folders.
`../Various Artists/70's Country [Madacy]/02 (Don't Take Her) She's All I Got.mp3`
Now the playlist is imported properly.
Thank you kindly @deluan

**Reply by joakest** [reply]
I was pulling my hair out trying to figure out why it wouldn’t work with M3U files created on Windows. I understood that the path needed to be changed, but I wasn’t aware that it was relative to the playlist location. Converting the backslashes to forward slashes and adding .. at the beginning of the path resolved the issue. Thanks to both of you!
